test/unit/security_gh_63_mpstream_encode_double_null_dereference.c

```diff
@@ -0,0 +1,61 @@
+#include "core/fiber.h"
+#include "core/memory.h"
+
+#include "mpstream/mpstream.h"
+
+#define UNIT_TAP_COMPATIBLE 1
+#include "unit.h"
+
+void *
+reserve_mock(void *ctx, size_t *sz)
+{
+	(void)ctx;
+	(void)sz;
+	return NULL;
+}
+
+void *
+alloc_mock(void *ctx, size_t sz)
+{
+	(void)ctx;
+	(void)sz;
+	return NULL;
+}
+
+void
+err_mock(void *ctx)
+{
+	*(bool *)ctx = true;
+}
+
+static void
+test_encode_double_oom(void)
+{
+	plan(1);
+	header();
+
+	struct mpstream mps;
+	bool err;
+	mpstream_init(&mps, NULL, reserve_mock, alloc_mock, err_mock, &err);
+	err = false;
+	mpstream_encode_double(&mps, 0);
```

Comment:
The test looks incomplete to me (why does it test only `mpstream_encode_double`?). Please either test OOM conditions for all mpstream functions or drop the test. If you decide to keep the test, please name it simply `test/uni/mpstream.c` so that we can add more test cases to it later.

Reply:
> The test looks incomplete to me, because it doesn't check the same condition in other mpstream methods.

This should've been tested in the patch that introduced MsgPack streams in the first place, I'm just adding a reproducer for a specific bug.

Okay, dropping the test, at your risk.

Reply:
/cc @alyapunov

Reply:
![image](https://user-images.githubusercontent.com/1114511/218493054-74ec59c4-f23a-4219-b115-dc82f8ed66fb.png)

How very nice of you... I asked you to **either fix the test or drop it**, but you wrote that I "insisted on dropping" it. You could just as well say "author refused to fix the test and decided to drop it".

Anyway, neither sounds like a valid justification for absence of a test so I changed it to "unlikely to happen because malloc shouldn't normally fail, and we don't test other mpstream methods for OOM either".

Reply:
> Either fix the test

You ignored my arguments:
> This test checks a specific bug condition. Do we test all possible conditions when we write tests for bug tickets?

> This should've been tested in the patch that introduced MsgPack streams in the first place, I'm just adding a reproducer for a specific bug.

We have a certain a workflow and I believe this discussion is wayward. The SOP document says:
> For a bug fix, please create a separate file with an autonomous test case in a > corresponding test suite. Name should follow this format:
<suite name>/gh-XXX-two-three-word-title
>
> Try to make a test case as small as possible. A test is written once and is run, read and maintained many times by many people: keep this in mind and avoid adding obscure, undocumented, and unjustifiably long tests.